pkg/config/yaml.go

```diff
@@ -350,40 +338,37 @@ func (d Dict) MarshalYAML() (interface{}, error) {
 	return g, nil
 }
 
-type yamlErrWithPos struct {
-	pos    Pos
-	errMsg string
-}
-
 // yaml.v3 errors are either TypeError - collection of error message or single error message.
 // Parse error messages to extract short error message and position.
-func parseYamlV3Error(err error) []yamlErrWithPos {
-	res := []yamlErrWithPos{}
+func parseYamlV3Error(err error) error {
+	errs := Errors{}
 	switch err := err.(type) {
 	case *yaml.TypeError:
 		for _, s := range err.Errors {
-			res = append(res, parseYamlV3ErrorString(s))
+			errs.Add(parseYamlV3ErrorString(s))
 		}
+	case PosError:
+		errs.Add(err)
 	default:
-		res = append(res, parseYamlV3ErrorString(err.Error()))
+		errs.Add(parseYamlV3ErrorString(err.Error()))
 	}
 
-	if len(res) == 0 { // should never happen
-		res = append(res, parseYamlV3ErrorString(err.Error()))
+	if !errs.Any() { // should never happen
+		errs.Add(parseYamlV3ErrorString(err.Error()))
 	}
-	return res
+	return errs
 }
 
 // parseYamlV3Error attempts to extract position and nice error message from yaml.v3 error message.
 // yaml.v3 errors are unstructured, use string parsing to extract information.
-// If no position can be extracted, returns (Pos{}, error.Error()).
+// If no position can be extracted, returns (Pos{}, error.Error()). !!!
```

Comment:
Why `!!!`?

Reply:
I use `!!!` as `"fix this before sending for review"` )